crates/ruff_python_ast/src/context.rs

```diff
@@ -300,6 +300,13 @@ impl<'a> Context<'a> {
             .map(|index| &self.scopes[*index])
     }
 
+    pub fn grandparent_scope(&self) -> Option<&Scope> {
```

Comment:
We should investigate why `nth(2)` doesn't give you the expected result. I recommend moving this function to your implementation if it is because of the way you traverse the AST.

Reply:
I modified it and instead get a copy of the scope stack which is the popped/pushed according to the scope.